http/src/main/java/io/micronaut/http/body/TextPlainHandler.java

```diff
@@ -44,26 +44,26 @@
 @Consumes(MediaType.TEXT_PLAIN)
 @Singleton
 @Experimental
-public final class TextPlainHandler implements MessageBodyHandler<String> {
+public final class TextPlainHandler implements MessageBodyHandler<CharSequence> {
```

Comment:
We might want to have an additional `MessageBodyHandler<String>` so the handler is found as a direct match.